store/hbase/snapshot.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hbasekv
+
+import (
+	"github.com/c4pt0r/go-hbase"
+	"github.com/pingcap/go-themis"
+	"github.com/pingcap/tidb/kv"
+	"github.com/juju/errors"
+)
+
+var (
+	_ kv.Snapshot = (*hbaseSnapshot)(nil)
+	_ kv.Iterator = (*hbaseIter)(nil)
+)
+
+type hbaseSnapshot struct {
+	txn       *themis.Txn
+	storeName string
+}
+
+func (s *hbaseSnapshot) Get(k kv.Key) ([]byte, error) {
+	g := hbase.NewGet([]byte(k))
+	g.AddColumn([]byte(ColFamily), []byte(Qualifier))
+	s.txn.Get(s.storeName, g)
+	r, err := s.txn.Get(s.storeName, g)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if r == nil {
+		return nil, kv.ErrNotExist
+	}
+	return r.Columns[ColFamily+":"+Qualifier].Value, nil
+}
+
+func (s *hbaseSnapshot) NewIterator(startKey interface{}) kv.Iterator {
+	scanner := s.txn.GetScanner([]byte(s.storeName), startKey.([]byte), nil)
+	it := &hbaseIter{
+		ThemisScanner: scanner,
+	}
+	r, _ := it.Next(nil)
```

Comment:
Check error, even if it isn't the return val.

r, err := it.Next()
if err != nil {
    log.Error(err)
}